src/main/java/seedu/address/logic/parser/FilterCommandParser.java

```diff
@@ -33,7 +34,9 @@ public FilterCommand parse(String args) throws ParseException {
         Lab lab = ParserUtil.parseLab(argMultimap.getValue(PREFIX_LAB).get());
         LabStatus labStatus = ParserUtil.parseLabStatus(argMultimap.getValue(PREFIX_LABSTATUS).get());
 
-        return new FilterCommand(new StudentHasLabPredicate(lab.of(labStatus)));
+        Lab labToFilter = lab.of(labStatus, new LabMark());
```

Comment:
i created another method `Lab#of(LabStatus, LabMark)` that checks the LabStatus and calls `Lab#of(LabMark)` if the labStatus is `GRADED` and `Lab#of(LabStatus)` otherwise. This is because it feels a bit weird to throw the responsibility of checking the labStatus in the FilterCommandParser.